backend/src/main/java/com/eyetrade/backend/service/TransactionService.java

```diff
@@ -101,11 +103,20 @@ public BuyTransactionResource buyFund(BuyTransactionDto transactionDto, UUID use
         return resource;
     }
 
+    @Transactional
+    public SellTransactionResource selfund(SellTransactionDto dto, UUID userId){
+        SellTransactionResource resource= new SellTransactionResource();
+        BuyTransactionResource buyTransactionResource=buyFund(new BuyTransactionDto(dto.getSoldTypeCurrency(),-dto.getAmount()),userId);
```

Comment:
Maybe we should add a control to prevent negative funds. For example if the user has 100 TL they can not sell 150 TL. However, I think it is not a big deal.

Reply:
nice point